### ISSUE TITLE
Linking against OpenCV

### ISSUE THREAD

**Issue by jpecar** [hpcugent/easybuild#293]
I'm trying to build something against EasyBuild provided OpenCV. I'm hitting this issue:
http://stackoverflow.com/questions/34401117/compiling-code-with-opencv-usr-bin-ld-cannot-find-lippicv

It looks like an issue with OpenCV: libippicv.a gets dumped under $EBROOTOPENCV/share/OpenCV/3rdparty/lib/libippicv.a where it has no chance of being picked up while  -lippicv is specified in opencv pkgconfig file.

How should EB handle this?

For now I'm just manually removing it from pkgconfig.

**Comment by boegel** [comment]
@jpecar What's the full output of `pkg-config opencv --cflags --libs`?

Isn't `pkg-config` supposed to inject the location to each of the libraries as `-L` flags?

It could be as simple as including `$EBROOTOPENCV/share/OpenCV/3rdparty/lib` in `$LIBRARY_PATH` though...

**Comment by jpecar** [comment]
pkg-config returns whatever is specified in $EBROOTOPENCV/lib/pkgconfig/opencv.pc ... and that 3rdparty path is not specified there. Therefore linking with libippicv fails.

I'm now removing -lippicv from opencv.pc ... as you mention, another option is to add that path to that file.

Where would one set proper LIBRARY_PATH? In the opencv module file?

**Comment by boegel** [comment]
@jpecar yes, you could do this by including the following in the OpenCV easyconfig file; the OpenCV module now already adds standard paths like `/lib` to `$LIBRARY_PATH`

```python
modextrapaths = {'LIBRARY_PATH': ['share/OpenCV/3rdparty/lib']}
```

Can you give this a try without patching `opencv.pc`, and see if that fixes your problem?

**Comment by jpecar** [comment]
Yep, that fixes it.
What is now the official procedure to introduce this line to opencv easyconfig file? :)

**Comment by boegel** [comment]
@jpecar Open a pull request to the `develop` branch of the https://github.com/hpcugent/easybuild-easyconfigs repository, ideally for all OpenCV easyconfigs.

You can use `eb` itself for that, see http://easybuild.readthedocs.io/en/latest/Integration_with_GitHub.html for more information.
